### PR TITLE
update-python-libraries: migrate git revs to tag

### PR DESCRIPTION
**Pull request by mweinelt**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Atemu — APPROVED** [review]
Not very familiar with this part of Nixpkgs but it looks sane to me at a glance.

**Review by philiptaron — APPROVED** [review]
I read the Python. 🐍

**Comment by mweinelt** [comment]
Hey, can we please give codeowners time to review and merge? Thank you.

**Comment by philiptaron** [comment]
> Hey, can we please give codeowners time to review and merge? Thank you.

My apologies for jumping the metaphorical gun.

**Comment by natsukium** [comment]
Thanks for working on and reviewing this change!
I just found [the context](https://github.com/NixOS/nixpkgs/pull/355973) for this PR. I think it's fine.

File: pkgs/by-name/up/update-python-libraries/update-python-libraries.py [hunk 1]
```diff
@@ -496,7 +496,7 @@ def _update_package(path, target):
         else:
             # forcefully rewrite rev, incase tagging conventions changed for a release
             match = matches[0]
-            text = text.replace(match, f'rev = "refs/tags/{prefix}${{version}}";')
+            text = text.replace(match, f'tag = "{prefix}${{version}}";')
```
Comment (mweinelt) [inline pkgs/by-name/up/update-python-libraries/update-python-libraries.py:499]: Seeing this now, we should unquote and uninterpolate the version, when the prefix is empty.

**Comment by zowoq** [comment]
@mweinelt Seems this blocking a lot of `nixpkgs-update` python packages updates, migrating the `src` to `tag` works but it then fails trying to eval `src.rev` in `meta.changelog`.

**Comment by Atemu** [comment]
That's a good time to stop doing that; you shouldn't assume anything about src other than that it's a path.

You might need to switch between rev and tag on updates or users may override src with anything they like.

Depending on src's internals feels like a layer violation to me.

**Comment by zowoq** [comment]
> That's a good time to stop doing that; you shouldn't assume anything about src other than that it's a path.

The assumption is that `meta` evaluates, using the same method that is enforced by nixpkgs CI.

**Comment by Atemu** [comment]
Sorry, I what I wrote was ambiguous: To stop depending on `src.tag` in meta; specifically depending on any internal state of `src` at all.

You'd necessarily need to pass the tag to the "constructor" of `src` anyways, so simply re-use that same declaration for changelog instead.

**Comment by zowoq** [comment]
Perhaps what I wrote was also ambiguous or maybe we're just talking past each other: nixpkgs-update doesn't depend on src for meta. The issue is that sometimes the migration of a python package from rev to tag is incomplete.

**Comment by Atemu** [comment]
Yes and the cause of that is the layer violation I talked about; depending on an internal property of `src`. If that property changes, breakage occurs. Therefore, you should not depend on the internal property.

**Comment by zowoq** [comment]
Yeah, I think we're just talking past each other.

**Comment by mweinelt** [comment]
> @mweinelt Seems this is blocking a lot of `nixpkgs-update` python packages updates, migrating the `src` to `tag` works but it then fails trying to eval `src.rev` in `meta.changelog`.

Thanks, I'll have a look in the next few days.

**Comment by mweinelt** [comment]
Not forgotten, but not so straightforward.

**Comment by mweinelt** [comment]
cb90424474c6ef5b7c00e5ddb4961647b73a043b

**Pull request merged**